### ISSUE TITLE
[pixman:x64-osx] Build fails

### ISSUE THREAD

**Issue by rozaxe** [Microsoft/vcpkg#3845]
Hi,

I am using macOS High Sierra 10.13.5 (fr-fr locale).

I tried to install `cairo` with `vcpkg install cairo` and the build fails when building `pixman` with this stacktrace :

```
[1/33] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -DPACKAGE=\"pixman-1\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_VERSION=\"0.34.0\" -DUSE_SSE2 -I/Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/. -g -MD -MT CMakeFiles/pixman-1.dir/pixman.c.o -MF CMakeFiles/pixman-1.dir/pixman.c.o.d -o CMakeFiles/pixman-1.dir/pixman.c.o   -c /Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/pixman.c
FAILED: CMakeFiles/pixman-1.dir/pixman.c.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -DPACKAGE=\"pixman-1\" -DPACKAGE_BUGREPORT=\"\" -DPACKAGE_VERSION=\"0.34.0\" -DUSE_SSE2 -I/Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/. -g -MD -MT CMakeFiles/pixman-1.dir/pixman.c.o -MF CMakeFiles/pixman-1.dir/pixman.c.o.d -o CMakeFiles/pixman-1.dir/pixman.c.o   -c /Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/pixman.c
In file included from /Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/pixman.c:29:
In file included from /Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/./pixman-private.h:34:
/Users/Axel/.build/vcpkg/buildtrees/pixman/src/pixman-0.34.0/pixman/./pixman-compiler.h:230:6: error: "Unknown thread local support for this system. Pixman will not work with multiple threads. Define PIXMAN_NO_TLS to acknowledge and accept this limitation and compile pixman without thread-safety support."
#    error "Unknown thread local support for this system. Pixman will not work with multiple threads. Define PIXMAN_NO_TLS to acknowledge and accept this limitation and compile pixman without thread-safety support."
     ^
1 error generated.
```

I expected to successfully install `cairo` (and thus, its dependency on `pixman`)

Apparently I need to define `PIXMAN_NO_TLS` somewhere, but even with `export PIXMAN_NO_TLS=1` in the terminal, the build fails.

[The full log install-x64-osx-dbg-out.log](https://github.com/Microsoft/vcpkg/files/2160650/install-x64-osx-dbg-out.log)

***
I also made sure I was running the last version of vcpkg with `git pull` and `vcpkg update`.

**Comment by angelmixu** [comment]
It happens the same to me, I can't get the macro being recognized when compiling.
I'm thinking that maybe, on macOS, we should use HAVE_PTHREADS as they're available on the system.

**Comment by ras0219-msft** [comment]
Should be fixed by #3923; thanks again, @angelmixu!
